src/TritonCTS/src/DbWrapper.cpp

```diff
@@ -219,19 +222,34 @@ void DbWrapper::computeITermPosition(odb::dbITerm* term, DBU &x, DBU &y) const {
 
 void DbWrapper::writeClockNetsToDb(const Clock& clockNet) {
         std::cout << " Writing clock net \"" << clockNet.getName() << "\" to DB\n";
+        odb::dbNet* topClockNet = _block->findNet(clockNet.getName().c_str());
+
+        // report clock sdc name
+        sta::dbNetwork* dbNetwork = _openSta->getDbNetwork();
+        odb::dbBTerm* clockBTerm = topClockNet->get1stBTerm();
+        if (clockBTerm != nullptr){
+                sta::ClockSet* clocksFound = _openSta->sdc()->findLeafPinClocks(dbNetwork->dbToSta(clockBTerm));
+                sta::ClockSet::Iterator clkIter(clocksFound);
+                while (clkIter.hasNext()) {
+                        sta::Clock* currentClk = clkIter.next();
+                        std::cout << " User-defined clock name: " << currentClk->name() << ".\n";
+                        break;
+                }
+        }
```

Comment:
Can this be done with a range-based for loop?

Reply:
I'll change it to a range-based loop in the next commit.
This uses a sta::set<sta::Clock*> to iterate over all sdc clocks connected to a pin. Perhaps @jjcherry56 knows a better way to find sdc clock names from a net/pin? Or is the current way ok?

Reply:
I can't tell what this code is actually trying to do but it looks very confused to me. The input arg is called "Clock" but it isn't an clock at all. It is the name of a clock net. We have gone over this before. CTS should not be using net names to represent clocks. It is a really bad design choice. It should be using sdc clocks. A clock has source pins that are connected to nets. The fanout of a clock source has multiple nets. So when you start with the wrong data structure you end up with convoluted bug prone code like this.  And guess what? This code assumes the clock is a top level port (BTerm). Which only works in some cases.

Reply:
TritonCTS uses an openSTA function (_openSta->findClkNets) to get all the dbNets that It works on, then, it creates a structure called "Clock" that only saves the names of these nets. (for now...)
Since I only had the name to work with, I created this function to get the SDC names based on the top-level port of the clock net returned by OpenSTA. 
Since this method has multiple problems, how should I rework TritonCTS to deal with sdc clocks instead of names? I'll type an example below:

The "Clock" class should be changed to use sdc::Clock and odb::dbNet as its only identifying structures. All subnets and buffers created should use dbNet and everything else sdc::Clock. This would mainly change the initClock function in TritonCTS.

However, is there a way to get the sdc::Clock from the nets of the _openSta->findClkNets functions? Or should this be done the other way around? (First get all sdc::Clocks, like I do in this segment of code, and then find their respective nets).

Thoughts?